javascript_nodejs/50.contoso-cafe-bot/dialogs/main/index.js

```diff
@@ -0,0 +1,260 @@
+                // No one has responded so start the right child dialog.
+                dialogTurnResult = await this.beginChildDialog(dc, onTurnProperty);
+            }
+
+            if(dialogTurnResult === undefined) return await dc.end();
+
+            // Examine result from dc.continue() or from the call to beginChildDialog().
+            switch (dialogTurnResult.status) {
+                case DialogTurnStatus.complete: {
+                    if (dialogTurnResult.result) {
+                        switch (dialogTurnResult.result.reason) {
+                            case 'Interruption': {
+                                // Interruption. Begin child dialog
+                                dialogTurnResult = await this.beginChildDialog(dc, onTurnProperty, dialogTurnResult.result.payload);
+                                break;
+                            } 
+                            case 'Abandon': {
+                                // Re-hydrate old dialog
+                                dialogTurnResult = await this.beginChildDialog(dc, dialogTurnResult.result.payload.onTurnProperty);
+                                break;
+                            }
+                        }
+                    } else {
+                        // The active dialog finished successfully. Ask user if they need help with anything else.
+                        await dc.context.sendActivity(MessageFactory.suggestedActions(GenSuggestedQueries(), `Is there anything else I can help you with?`));
+                        break;
+                    }
+                }
+                case DialogTurnStatus.waiting: {
+                    // The active dialog is waiting for a response from the user, so do nothing
+                    break;
+                }
+                case DialogTurnStatus.cancelled: {
+                    // The active dialog's stack has been cancelled
+                    await dc.context.sendActivity(MessageFactory.suggestedActions(GenSuggestedQueries(), `Is there anything else I can help you with?`));
+                    // End active dialog
+                    await dc.cancelAll();
+                    break;
+                }
+            }
+            dialogTurnResult = (dialogTurnResult === undefined) ? new TurnResultHelper(DialogTurnStatus.empty) : dialogTurnResult;
+            return dialogTurnResult;
+        }
+        /**
+         * Method to begin appropriate child dialog based on user input
+         * 
+         * @param {Object} dc 
+         * @param {Object} onTurnProperty 
+         * @param {Object} childDialogPayload 
+         */
+        async beginChildDialog(dc, onTurnProperty, childDialogPayload) {
+            switch (onTurnProperty.intent) {
+                // Help, ChitChat and QnA share the same QnA Maker model. So just call the QnA Dialog.
+                case QnADialog.Name: 
+                case ChitChatDialog.Name: 
+                case HelpDialog.Name: {
```

Comment:
Bracing on cases isn't necessary unless you need to create a block scope for `let` and `const`